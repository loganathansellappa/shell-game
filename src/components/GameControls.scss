.game-controls {
  display: flex;
  justify-content: space-around;
  text-decoration: none;
  color: white;
  cursor: pointer;

  .control {
    font-size: 1rem;
    max-width: 200px;
    border-radius: 15px;
  }
  .control-selected {
    background-color: #f9cd0b;
  }
}
