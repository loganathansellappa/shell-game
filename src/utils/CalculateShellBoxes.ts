import { Children } from 'react';

const calculateShellBoxes = (children) => {
  const boundingBoxes = {};

  Children.forEach(children, (child) => {
    const domNode = child.ref.current;
    if (domNode) {
      const nodeBoundingBox = domNode.getBoundingClientRect();

      boundingBoxes[child.key] = nodeBoundingBox;
    }
  });

  return boundingBoxes;
};

export default calculateShellBoxes;
