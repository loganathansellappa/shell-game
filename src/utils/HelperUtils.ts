import { ColorCodes } from '~/@types/GenericTypes';

export const shellCount = (): number => Number(process.env.SHELL_COUNT);

export const shuffle = <T>(elements: T[]): Array<T> => {
  for (let i = elements.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [elements[i], elements[j]] = [elements[j], elements[i]];
  }
  return [...elements];
};

export const getColors = (count: number): ColorCodes => {
  let colorCodes = {};
  for (let i = 0; i < count; i++) {
    colorCodes[`key${i}`] = `#${Math.floor(Math.random() * 16777215).toString(
      16
    )}`;
  }
  return colorCodes;
};

export enum Level {
  Beginner = 2,
  Intermediate = 4,
  Expert = 8,
}
