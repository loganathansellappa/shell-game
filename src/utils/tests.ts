import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};
export const testStore = () => {
  const mockStore = configureStore();
  const initialState = {
    gameLog: {
      gameCount: 1,
      winners: 1,
      losers: 0,
    },
  };
  const store = mockStore(initialState);
  return store;
};
