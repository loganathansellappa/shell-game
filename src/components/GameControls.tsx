import React, { forwardRef } from 'react';
import { Level } from '../utils/HelperUtils';
import './GameControls.scss';

interface GameControlsProps {
  startHandler: (hint?: boolean) => void;
  setGamerLevel: (shellId: number) => void;
  gamerLevel: Level;
  gameRunning: boolean;
}

export const GameControls = ({
  startHandler,
  setGamerLevel,
  gameRunning,
  gamerLevel,
}: GameControlsProps) => {
  console.log('gameRunning', gameRunning);
  console.log('gamerLevel', gamerLevel);
  return (
    <div className={'game-controls'}>
      <button
        disabled={gameRunning}
        className="control"
        onClick={() => startHandler()}
      >
        Start Game
      </button>
      <button
        disabled={gameRunning}
        className="control"
        onClick={() => startHandler(true)}
      >
        Start Game With Hint
      </button>
      <button
        disabled={gameRunning}
        className={`control ${
          gamerLevel === Level.Beginner ? 'control-selected' : ''
        }`}
        onClick={() => setGamerLevel(Level.Beginner)}
      >
        Beginner
      </button>
      <button
        disabled={gameRunning}
        className={`control ${
          gamerLevel === Level.Intermediate ? 'control-selected' : ''
        }`}
        onClick={() => setGamerLevel(Level.Intermediate)}
      >
        Intermediate
      </button>

      <button
        disabled={gameRunning}
        className={`control ${
          gamerLevel === Level.Expert ? 'control-selected' : ''
        }`}
        onClick={() => setGamerLevel(Level.Expert)}
      >
        Expert
      </button>
    </div>
  );
};
