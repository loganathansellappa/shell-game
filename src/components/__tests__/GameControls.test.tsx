import { GameControls } from '../GameControls';
import { renderWithRouter } from '../..//utils/tests';
import { Level } from '../../utils/HelperUtils';

test('renders GameControls Component', () => {
  const { container } = renderWithRouter(
    <GameControls
      startHandler={jest.fn()}
      setGamerLevel={jest.fn()}
      gamerLevel={Level.Beginner}
      gameRunning={false}
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
