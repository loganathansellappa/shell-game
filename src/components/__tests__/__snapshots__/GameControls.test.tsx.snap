// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GameControls Component 1`] = `
<div
  class="game-controls"
>
  <button
    class="control"
  >
    Start Game
  </button>
  <button
    class="control"
  >
    Start Game With Hint
  </button>
  <button
    class="control control-selected"
  >
    Beginner
  </button>
  <button
    class="control "
  >
    Intermediate
  </button>
  <button
    class="control "
  >
    Expert
  </button>
</div>
`;
